Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -355,35 +360,30 @@ void AnimationTimelinesController::registerNamedViewTimeline(const AtomString& n
     }
 }
 
-void AnimationTimelinesController::unregisterNamedViewTimelineForSubject(const AtomString& name, const Element& subject)
+void AnimationTimelinesController::unregisterNamedTimeline(const AtomString& name, const Element& element)
 {
-    auto it = m_nameToViewTimelinesMap.find(name);
-    if (it == m_nameToViewTimelinesMap.end())
+    auto it = m_nameToTimelineMap.find(name);
+    if (it == m_nameToTimelineMap.end())
         return;
-
-    auto& timelines = it->value;
-    timelines.removeFirstMatching([&](auto& timeline) {
-        return timeline->subject() == &subject;
+    it->value.removeFirstMatching([&] (auto& entry) {
+        return originatingElement(entry) == &element;
     });
-
-    if (timelines.isEmpty())
-        m_nameToViewTimelinesMap.remove(it);
+    if (it->value.isEmpty())
+        m_nameToTimelineMap.remove(name);
```

Comment:
I expect `m_nameToTimelineMap.remove(it)` is more efficient.

Reply:
Fixed.